test/e2e_test_attach.sh

```diff
@@ -46,6 +47,10 @@ IMAGE_URI_DIGEST=$IMAGE_URI@$SRC_DIGEST
 
 ## Sign with Leafcert Private Key
 openssl dgst -sha256 -sign ./private_key -out payload.sig payload.json
+## Generate TSR for the signature
+openssl ts -query -data payload.sig -sha256 -cert -out payload.tsq
+curl -H "Content-Type: application/timestamp-query" --data-binary '@payload.tsq' https://freetsa.org/tsr > payload.tsr
```

Comment:
Can we use the libraries in the sigstore/timestamp-authority repo to generate a timestamp (see unit tests, there should be some examples), or stand up an instance of the tsa for an e2e test (also there should be examples of such). We shouldn’t call out to a third party service as part of our tests.

Reply:
I tried to check the instance in the existing e2e test cases where we have used a separate instance of the tsa, but was not able to find any example. in the unit test i could see there are many instance of test cases covering the TSA testing and verification. so i think adding a new unit test case for attach case will not add up much functionality as in unit test case we directly modify the signature structure to include cert, cert-chain and tsa.
i was thinking if it is ok with you i can cerate a static TSR using freeTSA and use that instead of calling freeTSA in my test case. the certificates used  have a expiry time of 5 years so i can write a note in Readme.txt which suggest how to generate a fresh TSR when the existing certificates expires. please suggest what if this approach is ok

Reply:
This test has an example of spinning up a local server - https://github.com/sigstore/cosign/blob/main/test/e2e_test.go#L724

Reply:
Hi Hayden i have created a new test cases for using TSR in attach command in e2e_test.go file also i have reverted the changes in e2e_test_attach.sh file which were using freetsa for getting TSR. kindly review the changes,